src/main/java/io/codemc/bot/commands/CmdRemove.java

```diff
@@ -0,0 +1,122 @@
+ * and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.codemc.bot.commands;
+
+import com.jagrosh.jdautilities.command.SlashCommandEvent;
+import io.codemc.api.jenkins.JenkinsAPI;
+import io.codemc.api.nexus.NexusAPI;
+import io.codemc.bot.CodeMCBot;
+import io.codemc.bot.JavaContinuation;
+import io.codemc.bot.utils.CommandUtil;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.exceptions.ErrorHandler;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class CmdRemove extends BotCommand{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CmdRemove.class);
+
+    public CmdRemove(CodeMCBot bot){
+        super(bot);
+
+        this.name = "remove";
+        this.help = "Revoke a user's Author status.";
+
+        this.allowedRoles = bot.getConfigHandler().getLongList("allowed_roles", "commands", "service");
+
+        this.options = List.of(
+                new OptionData(OptionType.USER, "user", "The user that is having their status revoked.").setRequired(true),
+                new OptionData(OptionType.STRING, "username", "The Jenkins/Nexus username of the user. If blank, defaults to their server nickname.").setRequired(false)
+        );
+    }
+
+    @Override
+    public void withModalReply(SlashCommandEvent event){}
+
+    @Override
+    public void withHookReply(InteractionHook hook, SlashCommandEvent event, Guild guild, Member member){
+        Member user = event.getOption("user", OptionMapping::getAsMember);
+
+        if (user == null) {
+            CommandUtil.EmbedReply.from(hook).error("Invalid User was provided!").send();
+            return;
+        }
+
+        String username = event.getOption("username", user.getNickname(), OptionMapping::getAsString);

```

Comment:
As stated earlier, not a fan of having the Discord user be used as main source of getting the CI User. After all could their Discord name not match their GitHub name.

So if the above is changed should this here be altered too:
```suggestion
        String username = event.getOption("username", null, OptionMapping::getAsString);
        
        if (username == null || username.isEmpty()) {
            CommandUtil.EmbedReply.from(hook).error("Invalid CI User provided!").send();
            return;
        }
```